executor/src/witgen/column_map.rs

```diff
@@ -0,0 +1,95 @@
+use std::ops::{Index, IndexMut};
+
+use ast::analyzed::{PolyID, PolynomialType};
+
+/// A Map indexed by polynomial ID, for a specific polynomial type (e.g. fixed or witness).
+/// For performance reasons, it uses a Vec<V> internally and assumes that the polynomial IDs
+/// are contiguous.
```

Comment:
Does it break if they are not contiguous? I thought it would just be sparse but still work?

Reply:
You have to provide the range [0, max_id) and the initial values when you create a map.  So yeah, if you access an ID outside this range, this will break. Also, there is no notion of an "unset" key, if not all keys have values, you have to explicitly construct `ColumnMap<Option<V>>` (and can initialize with `None` as the initial value).

In the end, the `ColumnMap` just wraps a vector in places were a vector was used before (or when it's actually dense, like in `FixedData`).